plugins/by-name/treesitter/default.nix

```diff
@@ -366,24 +232,80 @@ lib.nixvim.plugins.mkNeovimPlugin {
   callSetup = false;
 
   extraConfig = cfg: opt: {
+    # TODO: Added 2025-12-18 Runtime detection to support legacy api
+    # Should be removed after transition period
+    # Runtime API detection and conditional setup
     plugins.treesitter.luaConfig.content =
-      # NOTE: Upstream state that the parser MUST be at the beginning of runtimepath.
-      # Otherwise the parsers from Neovim takes precedent, which may be incompatible with some queries.
-      (optionalString (cfg.settings.parser_install_dir != null) ''
-        vim.opt.runtimepath:prepend(${lib.nixvim.toLuaObject cfg.settings.parser_install_dir})
-      '')
-      + ''
-        require('nvim-treesitter.configs').setup(${lib.nixvim.toLuaObject cfg.settings})
+      let
+        # TODO: Added 2025-12-18 Check both legacy and new api options
+        # Add warning after transition period
+        # Map parser_install_dir to install_dir for main branch compatibility
+        # The main branch uses install_dir, while master branch uses parser_install_dir
+        mainBranchSettings =
+          lib.optionalAttrs ((cfg.settings.parser_install_dir or null) != null) {
+            install_dir = cfg.settings.parser_install_dir;
+          }
+          // lib.optionalAttrs (cfg.settings ? install_dir) {
+            inherit (cfg.settings) install_dir;
+          };
+
+        # TODO: Added 2025-12-18 Check both legacy and new api options
+        # Add warning after transition period
+        # Top-level options (cfg.highlight.enable, cfg.indent.enable) are for main branch
+        # For master branch compatibility, users should use settings.highlight.enable directly
+        highlightEnabled = cfg.highlight.enable || (cfg.settings.highlight.enable or false);
+        indentEnabled = cfg.indent.enable || (cfg.settings.indent.enable or false);
+
+        # TODO: Added 2025-12-18 Coerce string disable functions to rawLua
+        # Before it was dropped, the `highlight.disable` option did str→raw coercion
+        # Remove after transition period
+        coercedSettings =
+          cfg.settings
+          // lib.optionalAttrs (lib.isString (cfg.settings.highlight.disable or null)) {
+            highlight = cfg.settings.highlight // {
+              disable.__raw = cfg.settings.highlight.disable;
+            };
+          };
+      in
       ''
-      + (optionalString (cfg.languageRegister != { }) ''
-        do
-          local __parserFiletypeMappings = ${lib.nixvim.toLuaObject cfg.languageRegister}
+        -- Create autogroup for treesitter autocmds
+        local augroup = vim.api.nvim_create_augroup('nixvim_treesitter', { clear = true })
```

Comment:
It feels a little fragile that this assumes `plugins.treesitter.luaContent` will be _before_ `autoGroups`'s lua.

Maybe we should have a comment NOTEing that?

Reply:
I'm not sure I understand, this is the autoGroup lua. It's within treesitter luaContent. We put all plugin configs in `extraConfigLua` and the `autoCmd` stuff in `extraConfigLuaPost`. 

The current approach inlines everything into `plugins.treesitter.luaContent` except the one that will be added at bottom of file through the `autoCmd` option.

Reply:
The reason I say it is fragile is that we _also_ reference this autoGroup from the `autoCmd` option. If Nixvim gets an unrelated refactor in the future, assumptions about code order can break. I.e. the `autoCmd` option's lua may end up before the plugin's lua, causing it to be referencing an autogroup that doesn't exist yet.

Reply:
> Just felt like a concern in which that refactor would need to address / evaluate treewide effects of such a major change.

Maybe. But I'd rather avoid and/or highlight such fragile implementation details in advance, while we remember they exist, so that they become easier to spot later.

We've also done these kinda treewide changes _by accident_ before, without initially noticing that they broke things. E.g. when we introduced `plugins/by-name` and inadvertently changed the import order of almost all modules...